todo_app/app.py

```diff
@@ -1,54 +1,120 @@
 import os
+from flask_login.utils import login_required
 import requests
-from flask import Flask, render_template, request, session, redirect, url_for
+from oauthlib.oauth2 import WebApplicationClient
+from flask import Flask, render_template, request, session, redirect, url_for, abort, current_app
 import todo_app.view_model as view_model
 from todo_app.Task import Task
+from todo_app.user import User
 import todo_app.trello as trello
+from todo_app.mongo_db_tasks import TasksDb
+from flask_login import LoginManager, UserMixin, login_required, login_user, logout_user, current_user
 
-app = Flask(__name__)
-headers = {
-   "Accept": "application/json"
-}
+taskDb = TasksDb()
+secret_key = os.environ.get('SECRET_KEY', 'secret_key')
+git_client_id = os.environ.get('GIT_CLIENT_ID')
+git_client_secret = os.environ.get('GIT_CLIENT_SECRET')
+admin_user_id = os.environ.get('APP_ADMIN_USER_ID', '000')
+login_manager = LoginManager()
+client = WebApplicationClient(git_client_id)
+ROLE = 'readwrite'
 
-def create_app():
-    #app = Flask(__name__)
-    #app.config.from_object('dotenv')
+def create_app(taskDb):
+    app = Flask(__name__)    
+    login_disabled = os.environ.get('LOGIN_DISABLED', 'False') == 'True'
+    app.config['LOGIN_DISABLED'] = login_disabled
 
+    #login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():        
+        uri = client.prepare_request_uri("https://github.com/login/oauth/authorize")
+        return redirect(uri)
+
+    @login_manager.user_loader    
+    def load_user(user_id):        
+        return User(user_id)
+    
+    @app.route('/login/callback', methods=['GET'])
+    def login_callback():
+        token_response = requests.post('https://github.com/login/oauth/access_token', data={
+                'client_id': git_client_id,
+                'client_secret': git_client_secret,
+                'code': request.args.get('code')
+            }, headers={'Accept': 'application/json'})
+        
+        client.parse_request_body_response(token_response.text)
+
+        user_response = requests.get('https://api.github.com/user', headers={
+            'Accept': 'application/json', 
+            'Authorization': f"token {token_response.json()['access_token']}"
+            })
+        
+        login_user(User(user_response.json()['login']))        
+        return redirect(url_for('index'))
+    
+    def get_user_role():             
+        if (current_app.config.get('LOGIN_DISABLED') or current_user.user_id == admin_user_id):                        
+            return ROLE
+        else:            
+            return 'readonly'
```

Comment:
Doing the check here for LOGIN_DISABLED makes sense, but otherwise it feels like the role would be better suited as a property on the User class